### ISSUE TITLE
The `adapter-ip` setting is not preserved in paused.conf

### ISSUE THREAD

**Issue by mzpqnxow** [robertdavidgraham/masscan#495]
It appears that as of the latest build from master, masscan doesn't save the `adapter-ip` setting in paused.conf. It's not difficult to manually add it back into the file yourself (if you happen to know/remember it) and then continue on using `--resume`, but this is obvious a bug that needs to be fixed

I will send a PR if I have time (I may have code from a prior bug with a different variable somewhere) but if someone else wants to take this, please do!

Thanks

**Comment by mzpqnxow** [comment]
So I just took a quick look at the code for this, here is what I see:


```

/***************************************************************************
 * Echoes the configuration for one nic
 ***************************************************************************/
static void
masscan_echo_nic(struct Masscan *masscan, FILE *fp, unsigned i)
{
    char zzz[64];

    /* If we have only one adapter, then don't print the array indexes.
     * Otherwise, we need to print the array indexes to distinguish
     * the NICs from each other */
    if (masscan->nic_count <= 1)
        zzz[0] = '\0';
    else
        sprintf_s(zzz, sizeof(zzz), "[%u]", i);

    if (masscan->nic[i].ifname[0])
        fprintf(fp, "adapter%s = %s\n", zzz, masscan->nic[i].ifname);
    
    if (masscan->nic[i].src.ip.first+1 == masscan->nic[i].src.ip.last)
        fprintf(fp, "adapter-ip%s = %u.%u.%u.%u\n", zzz,
            (masscan->nic[i].src.ip.first>>24)&0xFF,
            (masscan->nic[i].src.ip.first>>16)&0xFF,
            (masscan->nic[i].src.ip.first>> 8)&0xFF,
            (masscan->nic[i].src.ip.first>> 0)&0xFF
            );
    else if (masscan->nic[i].src.ip.first+1 < masscan->nic[i].src.ip.last)
        fprintf(fp, "adapter-ip%s = %u.%u.%u.%u-%u.%u.%u.%u\n", zzz,
            (masscan->nic[i].src.ip.first>>24)&0xFF,
            (masscan->nic[i].src.ip.first>>16)&0xFF,
            (masscan->nic[i].src.ip.first>> 8)&0xFF,
            (masscan->nic[i].src.ip.first>> 0)&0xFF,
            (masscan->nic[i].src.ip.last>>24)&0xFF,
            (masscan->nic[i].src.ip.last>>16)&0xFF,
            (masscan->nic[i].src.ip.last>> 8)&0xFF,
            (masscan->nic[i].src.ip.last>> 0)&0xFF
            );

#ifdef MZPWNXOW_TEST
    else
        abort();
#endif 
```

Note the code with the `#ifdef` is what I just added.

Sure enough, it aborts when I run with very basic arguments and then trigger creation of paused.conf with a control-c:

```
$ masscan --adapter-ip 1.2.3.4 1.1.1.1 -p 1-64454 --interface eth0 --router-mac 44:44:44:44:44:44
[1]    2431 abort
```

The logic appears to be present for handling cases where there are multiple --adapter-ip values. I didn't know this was supported, but apparently it is. In any even, I found that swapping around the if and making the second one an `else` instead of an `else if`, `adapter-ip` is in-fact written to `paused.conf`. This is the modified version that works for me:

```
    if (masscan->nic[i].src.ip.first+1 < masscan->nic[i].src.ip.last)
        fprintf(fp, "adapter-ip%s = %u.%u.%u.%u-%u.%u.%u.%u\n", zzz,
            (masscan->nic[i].src.ip.first>>24)&0xFF,
            (masscan->nic[i].src.ip.first>>16)&0xFF,
            (masscan->nic[i].src.ip.first>> 8)&0xFF,
            (masscan->nic[i].src.ip.first>> 0)&0xFF,
            (masscan->nic[i].src.ip.last>>24)&0xFF,
            (masscan->nic[i].src.ip.last>>16)&0xFF,
            (masscan->nic[i].src.ip.last>> 8)&0xFF,
            (masscan->nic[i].src.ip.last>> 0)&0xFF
            );
    else
        fprintf(fp, "adapter-ip%s = %u.%u.%u.%u\n", zzz,
            (masscan->nic[i].src.ip.first>>24)&0xFF,
            (masscan->nic[i].src.ip.first>>16)&0xFF,
            (masscan->nic[i].src.ip.first>> 8)&0xFF,
            (masscan->nic[i].src.ip.first>> 0)&0xFF
            );
```

I'll have to look into the multiple `--adapter-ip` case as obviously I don't want to break that, but maybe this will Just Work(TM)

**Comment by mzpqnxow** [comment]
OK, I think I understand the multiple `adapter-ip` feature now, the syntax is:

```
--adapter-ip 1.2.3.0-1.2.3.3
```
I confirmed that changing the code as I did above results in the same output when using the multiple `--adapter-ip` mode, and also causes the correct output for a single `--adapter-ip`, so I think I will send this as a PR

**Comment by mzpqnxow** [comment]
After a more careful look at this, I submitted #496 which fixes two slightly different bugs in the same function. I also took the opportunity to rename the `xxx` variable to something less nonsensical :)

Applying this patch allows the following to be correctly written to paused.conf. Without it, neither of the following invocations would write any `adapter-ip` setting to `paused.conf`, causing a resumed scan to fail when explicitly specifying `adapter-ip` was required/desired

`masscan ... --adapter-ip 1.2.3.1-1.2.3.3` - Failed due to off by one
`masscan ... --adapter-ip 1.2.3.1` - Failed due to off by one that looks more like a typo/copy paste

Only ranges of size 3 or more worked (e.g. `masscan ... --adapter-ip 1.2.3.1-1.2.3.4`
